axelrod/tournament.py

```diff
@@ -234,20 +244,18 @@ def _write_interactions_to_dict(self, results):
                     self.interactions_dict[index_pair] = [interaction]
                 self.num_interactions += 1
 
-    def _run_parallel(self, processes: int=2, progress_bar: bool = False
-                      ) -> bool:
+    def _run_parallel(self, processes: int=2) -> bool:
         """
         Run all matches in parallel
 
         Parameters
         ----------
 
-        progress_bar : bool
-            Whether or not to update the tournament progress bar
+        processes : int
+            How many processes to use.
         """
         # At first sight, it might seem simpler to use the multiprocessing Pool
```

Comment:
tl;dr - pickling bound methods supported since py3.4 .  my test on axelrod confirmed with py3.5.3. other bound method tests confirmed with 3.4 and 3.5.2 (i can't axelrod below 3.5.3 for ... reasons)

fun reading [here](https://bugs.python.org/issue9276)    

on a different test, i get interesting results:
```
import pickle

class Tourney(object):
    def __init__(self, a_list):
        self.lst = a_list

    def play(self):
        return self.lst.pop()

    @staticmethod
    def hi():
        return 'hi'

if __name__ == '__main__':
    tournament = Tourney([1, 2, 3])

    bm_1 = tournament.play
    print('original bound method: ', bm_1)
    result_set_1 = bm_1()

    bm_2 = pickle.loads(pickle.dumps(bm_1))
    print('second bound method: ', bm_2)
    result_set_2 = bm_2()

    bm_3 = pickle.loads(pickle.dumps(bm_2))
    print('third bound method: ', bm_3)
    result_set_3 = bm_3()

    print(result_set_1, result_set_2, result_set_3)

    um = Tourney.hi
    print(um)
    print(pickle.loads(pickle.dumps(um))())
```

on py3.5.2 this works as expected, but it fails on py3.4.4.  unbound methods were briefly un-pickle-able.

```
D:\PROGRAMS\WinPython-64bit-3.4.4.4Qt5\python-3.4.4.amd64\python.exe C:/Users/Administrator/.PyCharm2017.1/config/scratches/scratch_6.py
Traceback (most recent call last):
original bound method:  <bound method Tourney.play of <__main__.Tourney object at 0x0000000001F62208>>
  File "C:/Users/Administrator/.PyCharm2017.1/config/scratches/scratch_6.py", line 37, in <module>
    print(pickle.loads(pickle.dumps(um))())
second bound method:  <bound method Tourney.play of <__main__.Tourney object at 0x0000000001F622B0>>
_pickle.PicklingError: Can't pickle <function Tourney.hi at 0x0000000001FAC2F0>: attribute lookup hi on __main__ failed
third bound method:  <bound method Tourney.play of <__main__.Tourney object at 0x0000000001F626A0>>
3 2 1
False
<function Tourney.hi at 0x0000000001FAC2F0>

Process finished with exit code 1
```